vnext/JSI/Shared/ChakraApi.cpp

```diff
@@ -0,0 +1,439 @@
+}
+
+JsPropertyIdType ChakraApi::GetPropertyIdType(JsPropertyIdRef propertyId) {
+  JsPropertyIdType type;
+  ChakraVerifyJsErrorElseThrow(JsGetPropertyIdType(propertyId, &type));
+  return type;
+}
+
+JsPropertyIdRef ChakraApi::GetPropertyIdFromSymbol(JsValueRef symbol) {
+  JsPropertyIdRef propertyId{JS_INVALID_REFERENCE};
+  ChakraVerifyJsErrorElseThrow(JsGetPropertyIdFromSymbol(symbol, &propertyId));
+  return propertyId;
+}
+
+JsPropertyIdRef ChakraApi::GetPropertyIdFromSymbol(std::wstring_view symbolDescription) {
+  JsPropertyIdRef propertyId;
+  ChakraVerifyJsErrorElseThrow(JsGetPropertyIdFromSymbol(CreateSymbol(symbolDescription), &propertyId));
+  return propertyId;
+}
+
+JsValueRef ChakraApi::CreateSymbol(JsValueRef symbolDescription) {
+  JsValueRef symbol;
+  ChakraVerifyJsErrorElseThrow(JsCreateSymbol(symbolDescription, &symbol));
+  return symbol;
+}
+
+JsValueRef ChakraApi::CreateSymbol(std::wstring_view symbolDescription) {
+  return CreateSymbol(PointerToString(symbolDescription));
+}
+
+JsValueRef ChakraApi::GetUndefinedValue() {
+  JsValueRef undefinedValue;
+  ChakraVerifyJsErrorElseThrow(JsGetUndefinedValue(&undefinedValue));
+  return undefinedValue;
+}
+
+JsValueRef ChakraApi::GetNullValue() {
+  JsValueRef nullValue;
+  ChakraVerifyJsErrorElseThrow(JsGetNullValue(&nullValue));
+  return nullValue;
+}
+
+JsValueRef ChakraApi::BoolToBoolean(bool value) {
+  JsValueRef booleanValue;
+  ChakraVerifyJsErrorElseThrow(JsBoolToBoolean(value, &booleanValue));
+  return booleanValue;
+}
+
+bool ChakraApi::BooleanToBool(JsValueRef value) {
+  bool boolValue;
+  ChakraVerifyJsErrorElseThrow(JsBooleanToBool(value, &boolValue));
+  return boolValue;
+}
+
+JsValueType ChakraApi::GetValueType(JsValueRef value) {
+  JsValueType type;
+  ChakraVerifyJsErrorElseThrow(JsGetValueType(value, &type));
+  return type;
+}
+
+JsValueRef ChakraApi::DoubleToNumber(double value) {
+  JsValueRef result;
+  ChakraVerifyJsErrorElseThrow(JsDoubleToNumber(value, &result));
+  return result;
+}
+
+JsValueRef ChakraApi::IntToNumber(int32_t value) {
+  JsValueRef numberValue;
+  ChakraVerifyJsErrorElseThrow(JsIntToNumber(value, &numberValue));
+  return numberValue;
+}
+
+double ChakraApi::NumberToDouble(JsValueRef value) {
+  double doubleValue;
+  ChakraVerifyJsErrorElseThrow(JsNumberToDouble(value, &doubleValue));
+  return doubleValue;
+}
+
+int32_t ChakraApi::NumberToInt(JsValueRef value) {
+  int intValue;
+  ChakraVerifyJsErrorElseThrow(JsNumberToInt(value, &intValue));
+  return intValue;
+}
+
+JsValueRef ChakraApi::PointerToString(std::wstring_view value) {
+  ChakraVerifyElseThrow(value.data(), "Cannot convert a nullptr to a JS string.");
+
+  JsValueRef result{JS_INVALID_REFERENCE};
+  ChakraVerifyJsErrorElseThrow(JsPointerToString(value.data(), value.size(), &result));
+  return result;
+}
+
+JsValueRef ChakraApi::PointerToString(std::string_view value) {
+  ChakraVerifyElseThrow(value.data(), "Cannot convert a nullptr to a JS string.");
+
```

Comment:
can't/shouldn't we convert it to the js value `null`?

Reply:
No, it is better not, because the function contract says that we return a string object. It may cause unexpected issues in the code that consumes it.

---
In reply to: [470463800](https://github.com/microsoft/react-native-windows/pull/5710#discussion_r470463800) [](ancestors = 470463800)